src/queryValidator.js

```diff
@@ -0,0 +1,67 @@
+import { keyTypes } from "./constants.js";
+import { createError } from "./error.js";
+
+export const queryValidator = (queryString) => {
+  const error = [];
+
+  // Check if the URL is encoded and if there is something before CMCD request
+  if (decodeURI(queryString) === queryString) {
+    error.push(createError("parameter-encoding"));
+  } else {
+    const query = queryString.split("?").pop();
+
+    //Check if the query is correct
+    const requests = decodeURIComponent(query).split("CMCD=");
+    requests.shift();
+
+    if (requests.length > 1) {
+      error.push(createError("incorrect-format"));
+    } else {
+      const values = decodeURIComponent(query).split("CMCD=")[1].split(",");
+      values[values.length - 1] = values[values.length - 1].split("&")[0];
+
+      // console.log("values\n", values);
+
+      const keys = []
+
+      // Check: key/value is separated by =
+      for (const val of values) {
+        const [key, value] = val.split("=");
+        keys.push(key)
+
+        //Check: string require ""
+        if (
+          (keyTypes[key] === "String" &&
+            !(
+              value.indexOf('"') == 0 &&
+              value.lastIndexOf('"') == value.length - 1
+            )) ||
+          (keyTypes[key] === "Token" &&
+            value.indexOf('"') == 0 &&
+            value.lastIndexOf('"') == value.length - 1)
```

Comment:
let's abstract this check to a new method, so that it's clearer to read. Also I believe we can reduce this logic since the second condition && is repeated.

Reply:
I agree with this.
Also, maybe we can abstract the types of keys to an enum, so that the use is something like:
```
keyTypes[key] === KeyType.String
keyTypes[key] === KeyType.Token
```

This way, we avoid human error typos in type checking. Also, should be easy to change/update the value of the key type in the future if needed.

Though, maybe it's not needed/wanted for Milestone 1, but create the ticket to reflect this change.

Reply:
I made some changes in constants.js regarding this. @mpollingerQualabs and @arrivillagaTomas Could you take a look into it?